File: 1578-minimum-time-to-make-rope-colorful/1578-minimum-time-to-make-rope-colorful.py
class Solution:
    def minCost(self, s: str, neededTime: List[int]) -> int:
        ans = left = 0 
        while left < len(s):
            maxEffort = totalEffort = neededTime[left]
            right = left + 1
            while right < len(neededTime) and s[right] == s[left]:
                totalEffort += neededTime[right]
                maxEffort = max(maxEffort, neededTime[right])
                right += 1
            ans += (totalEffort - maxEffort)
            left = right
            
        return ans
            
            

File: 1578-minimum-time-to-make-rope-colorful/1578-minimum-time-to-make-rope-colorful.cpp
class Solution {
public:
    int minCost(string s, vector<int>& neededTime) {
        int ans = 0, i = 0;
        
        while (i < neededTime.size()){
            int maxEffort = neededTime[i], totalEffort =neededTime[i];
            int j = i+1;
            bool isEffortRequired = false;
            
            while(j < neededTime.size() && s[i] == s[j]){
                isEffortRequired = true;
                totalEffort += neededTime[j];
                maxEffort = max(maxEffort, neededTime[j]);
                j += 1;
            }
            
            if (isEffortRequired){
                ans += (totalEffort - maxEffort);
            }
            
            i = j;
            
        }
        return ans;
        
        
        
        
    }
};